lib/fastlane/plugin/wpmreleasetoolkit/versioning/formatters/derived_build_code_formatter.rb

```diff
@@ -67,6 +79,35 @@ def validate_prefix!(prefix)
 
           UI.user_error!("Prefix must be an integer digit (0-9) or empty string, got: '#{prefix_str}'")
         end
+
+        # Validates that the digit count is a valid positive integer within reasonable limits.
+        #
+        # @param [Integer] digit_count The digit count to validate
+        #
+        # @raise [StandardError] If the digit count is invalid
+        #
+        def validate_digit_count!(digit_count)
+          # Check if it's an integer
+          unless digit_count.is_a?(Integer)
+            UI.user_error!("Digit count must be an integer, got: #{digit_count.class}")
+          end
+
+          return if digit_count.between?(MIN_DIGIT_COUNT, MAX_DIGIT_COUNT)
+
+          UI.user_error!("Digit count must be between #{MIN_DIGIT_COUNT} and #{MAX_DIGIT_COUNT} digits, got: #{digit_count}")
+        end
+
+        # Validates that the total number of digits (excluding prefix) doesn't exceed the maximum for multiplatform compatibility.
+        #
```

Comment:
💄  Adding extra context in comment about the reason us wanting to limit the total number of digits being about Google Play's ≈`2_000_000_000` limit.
```suggestion
        # Validates that the total number of digits (excluding prefix) doesn't exceed the maximum for multiplatform compatibility.
        #
        # Since Google Play's max versionCode is ≈ 2_000_000_000, we want to avoid being too close to the limit
        # as this would then block us from submitting any updates for that app if we reached it.
```

Reply:
Good one 👍